include/aspect/volume_of_fluid/utilities.h

```diff
@@ -0,0 +1,152 @@
+
+    /**
+     * Obtain values at points for a polynomial function that is equivalent to
+     * the Heaviside function $H(d-normal*xhat)$ on the unit cell when
+     * integrated against polynomials of up to the specified degree.
+     *
+     * Currently works for degree <=1
+     *
+     * @param degree Maximum degree for exact integration
+     * @param normal Interface normal vector, pointed away from the included region
+     * @param d Interface parameter specifying location of interface in unit cell
+     * @param points Locations to evaluate the constructed polynomial
+     * @param values Values of the constructed polynomial at the specified points
+     */
+    template<int dim>
+    void xFEM_Heaviside(const int degree,
+                        const Tensor<1, dim, double> normal,
+                        const double d,
+                        const std::vector<Point<dim>> &points,
+                        std::vector<double> &values);
+    /**
+     * Obtain values at points for a polynomial function that is equivalent to
+     * the function $\frac{d}{dd}H(d-normal*xhat)$ on the unit cell when
+     * integrated against polynomials of up to the specified degree.
+     *
+     * Currently works for degree <=1
+     *
+     * @param degree Maximum degree for exact integration
+     * @param normal Interface normal vector, pointed away from the included region
+     * @param d Interface parameter specifying location of interface in unit cell
+     * @param points Locations to evaluate the constructed polynomial
+     * @param values Values of the constructed polynomial at the specified points
+     */
+    template<int dim>
+    void xFEM_Heaviside_derivative_d(const int degree,
+                                     const Tensor<1, dim, double> normal,
+                                     const double d,
+                                     const std::vector<Point<dim>> &points,
+                                     std::vector<double> &values);
+
+
+    /**
+     * Function to do Newton iteration calculation of correct d for a given
+     * normal to get vol_frac from xFEM_Heaviside integrated against the given
+     * weights.
+     *
+     * @param degree Maximum degree for exact integration
+     * @param normal Interface normal vector, pointed away from the included region
+     * @param vol_frac Cell volume fraction in physical space - used for target value for integration
+     * @param vol Cell volume in physical space - used for target value for integration
+     * @param epsilon Tolerance for Newton iteration
+     * @param points Quadrature points to use for update
+     * @param weights JxW values to use for quadrature
+     */
+    template<int dim>
+    double compute_interface_location_newton(const int degree,
+                                             const Tensor<1, dim, double> normal,
+                                             const double vol_frac,
```

Comment:
Please replace any remaining location of `vol_frac` with `volume_fraction`, the few saved characters are not worth the mental effort to replace this every time someone reads it.